src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -671,20 +732,52 @@ async def get_workspace_managers(
-                UserAppletAccessSchema.role != Role.RESPONDENT,
                 has_access,
+                UserAppletAccessSchema.role != Role.RESPONDENT,
                 UserAppletAccessSchema.applet_id == applet_id if applet_id else True,
             )
-            .group_by(UserSchema.id, UserAppletAccessSchema.title)
+            .group_by(UserSchema.id)
+        ).cte("accepted_users")
+
+        query: Query = (
+            select(
+                func.coalesce(accepted_users.c.id, invited_users.c.id).label("id"),
+                func.coalesce(accepted_users.c.first_name, invited_users.c.first_name).label("first_name"),
+                func.coalesce(accepted_users.c.last_name, invited_users.c.last_name).label("last_name"),
+                func.coalesce(accepted_users.c.email_encrypted, invited_users.c.email).label("email_encrypted"),
+                func.coalesce(accepted_users.c.created_at, invited_users.c.created_at).label("created_at"),
+                func.coalesce(accepted_users.c.last_seen, invited_users.c.created_at).label("last_seen"),
+                func.coalesce(accepted_users.c.roles, invited_users.c.roles).label("roles"),
+                func.coalesce(accepted_users.c.applets, invited_users.c.applets, []).label("applets"),
+                func.coalesce(accepted_users.c.titles, invited_users.c.titles, []).label("titles"),
+                func.coalesce(invited_users.c.status, InvitationStatus.APPROVED).label("status"),
+                func.coalesce(accepted_users.c.is_pinned, False).label("is_pinned"),
+            )
+            .select_from(invited_users)
+            .join(accepted_users, accepted_users.c.id == invited_users.c.user_id, isouter=True, full=True)
         )
 
         if query_params.filters:
             query = query.where(*_AppletUsersFilter().get_clauses(**query_params.filters))
 
         # Get total count before ordering to evaluate eligibility for ordering by encrypted fields
-        coro_total = self._execute(select(count()).select_from(query.with_only_columns(UserSchema.id).subquery()))
+        coro_total = self._execute(
+            select(count()).select_from(query.with_only_columns(literal_column("status")).subquery())
+        )
         total = (await coro_total).scalar()
 
+        class _AppletManagersOrdering(Ordering):
+            created_at = Ordering.Clause(literal_column("created_at"))
+            is_pinned = Ordering.Clause(literal_column("is_pinned"))
+            last_seen = Ordering.Clause(literal_column("last_seen"))
+            roles = Ordering.Clause(literal_column("roles"))
+
+            encrypted_fields = {
+                "email": Ordering.Clause(func.decrypt_internal(accepted_users.c.email_encrypted, get_key())),
+                "first_name": Ordering.Clause(func.decrypt_internal(accepted_users.c.first_name, get_key())),
+                "last_name": Ordering.Clause(func.decrypt_internal(accepted_users.c.last_name, get_key())),
```

Comment:
Because these ordering fields are only referencing the `accepted_users` results, it causes ordering to always first be grouped by status, then secondly by the requested ordering field. See demo recording below (I quickly added the `status` column to the Team page for illustration):

https://github.com/ChildMindInstitute/mindlogger-backend-refactor/assets/1282772/155a2d9f-0764-450b-b2c3-e344ff7f01d4

However, if we use the same `func.coalesce` approach to pick the correct field (as is used in the `query`'s `select` call), we get the desired sorting behaviour. (I tried using `literal_column(…)` instead of repeating `func.coalesce(…)`, but Postgres didn't like it!)

```suggestion
                "email": Ordering.Clause(
                    func.decrypt_internal(
                        func.coalesce(accepted_users.c.email_encrypted, invited_users.c.email), get_key()
                    )
                ),
                "first_name": Ordering.Clause(
                    func.decrypt_internal(
                        func.coalesce(accepted_users.c.first_name, invited_users.c.first_name), get_key()
                    )
                ),
                "last_name": Ordering.Clause(
                    func.decrypt_internal(
                        func.coalesce(accepted_users.c.last_name, invited_users.c.last_name), get_key()
                    )
                ),
```

See working demo:

https://github.com/ChildMindInstitute/mindlogger-backend-refactor/assets/1282772/73325e0a-7966-4421-b359-be0ee3812bed

Reply:
great!! thanks

Reply:
done!